mm2src/mm2_main/src/lp_swap/swap_watcher.rs

```diff
@@ -152,17 +155,38 @@ impl State for ValidateTakerFee {
     type Result = ();
 
     async fn on_changed(self: Box<Self>, watcher_ctx: &mut WatcherContext) -> StateResult<Self::Ctx, Self::Result> {
```

Comment:
Could you check the following diff please?
```diff
diff --git a/mm2src/mm2_main/src/lp_swap/swap_watcher.rs b/mm2src/mm2_main/src/lp_swap/swap_watcher.rs
index 3401244e0..a52665d4c 100644
--- a/mm2src/mm2_main/src/lp_swap/swap_watcher.rs
+++ b/mm2src/mm2_main/src/lp_swap/swap_watcher.rs
@@ -155,6 +155,8 @@ impl State for ValidateTakerFee {
     type Result = ();

     async fn on_changed(self: Box<Self>, watcher_ctx: &mut WatcherContext) -> StateResult<Self::Ctx, Self::Result> {
+        const MAX_ATTEMPT_LIMIT: u8 = 3;
+
         let taker_amount = MmNumber::from(watcher_ctx.data.taker_amount.clone());
         let fee_amount =
             dex_fee_amount_from_taker_coin(&watcher_ctx.taker_coin, &watcher_ctx.data.maker_coin, &taker_amount);
@@ -176,18 +178,21 @@ impl State for ValidateTakerFee {
                 .await
             {
                 Ok(_) => break,
-                Err(err) => {
-                    if attempts >= 3 {
-                        return Self::change_state(Stopped::from_reason(StopReason::Error(
-                            WatcherError::InvalidTakerFee(format!("{:?}", err)).into(),
-                        )));
-                    } else {
-                        attempts += 1;
-                        Timer::sleep(10.).await;
-                    }
+                Err(err) if attempts >= MAX_ATTEMPT_LIMIT => {
+                    // please inform developer with debug log here
+                    return Self::change_state(Stopped::from_reason(StopReason::Error(
+                        WatcherError::InvalidTakerFee(format!("{:?}", err)).into(),
+                    )));
+                },
+                Err(_) => {
+                    // please inform developer with debug log here
+                    attempts += 1;
+                    Timer::sleep(10.).await;
                 },
             };
         }
+
+        // please inform user with info log here
         Self::change_state(ValidateTakerPayment {})
     }
 }
```

Reply:
Actually after some consideration I realized that attempting to validate the taker fee 4 times is not really necessary for watchers, so I'll remove the whole loop.